spec/GraphQLOverHTTP.md

```diff
@@ -237,120 +311,304 @@ omitted.
 
 GET requests MUST NOT be used for executing mutation operations. If the values
 of {query} and {operationName} indicate that a mutation operation is to be
-executed, the server SHOULD immediately respond with error status code `405`
-(Method Not Allowed) and halt execution. This restriction is necessary to
-conform with the long-established semantics of safe methods within HTTP.
+executed, the server MUST respond with error status code `405` (Method Not
+Allowed) and halt execution. This restriction is necessary to conform with the
+long-established semantics of safe methods within HTTP.
```

Comment:
Would it be interesting to include a further non-normative note here saying something like:

Additionally, restricting mutations to POSTs which MUST contain a `Content-Type` header ensures that mutations cannot be sent as part of [cross-site request forgery (CSRF) attacks](https://owasp.org/www-community/attacks/csrf), as long as the `Content-Type` header is not allowed to be one of the three values which can be in a [CORS-safelisted `Content-Type` request header](https://fetch.spec.whatwg.org/#cors-safelisted-request-header) (`application/x-www-form-urlencoded`, `multipart/form-data`, or `text/plain`).

(that `MUST` assumes my other suggestion is taken, which perhaps it won't be)